planb/main.m

```diff
@@ -50,18 +69,32 @@
   E.g. using the default parameters above with the parameter 'pkg1/sample' the URL would be:
   https://mac.internal.megacorp.com/pkgbase/pkg1/sample-stable.dmg
 */
-NSURL* URLForPackagePath(NSString *pkg) {
-  static dispatch_once_t onceToken;
-  static NSString *track;
+NSString* StringURLForPackagePath(NSString *pkg) {
+  NSString *path = [NSString stringWithFormat:@"%@-%@.dmg", pkg, MachineTrack()];
+  NSURL *url = [NSURL URLWithString:path relativeToURL:[NSURL URLWithString:kBaseURL]];
+  return [url absoluteString];
+}
 
-  dispatch_once(&onceToken, ^{
-    NSDictionary *machineInfoDictionary = [NSDictionary dictionaryWithContentsOfFile:kMachineInfo];
-    track = [[machineInfoDictionary objectForKey:kMachineInfoKey] lowercaseString];
-    if (!track.length) track = @"stable";
-  });
+/**
+ Return the list of packages to install, both static and from a downloaded manifest.
+ */
+NSArray* Packages(NSURLSession *session) {
+  NSMutableArray *result = [NSMutableArray array];
+
+  for (NSArray *item in StaticPackages()) {
+    NSMutableArray *newItem = [NSMutableArray arrayWithArray:item];
+    newItem[0] = StringURLForPackagePath(item[0]);
+    [result addObject:newItem];
+  }
+
+  if (kManifestURL.length) {
+    PBManifest *manifest = [[PBManifest alloc] initWithURL:[NSURL URLWithString:kManifestURL]];
+    if (session) manifest.session = session;
```

Comment:
if (session) is not needed

Reply:
I think this check is so that we keep the default [NSURLSession sharedSession] that manifest.session is initialized with if session is nil.  Not sure if that is a good idea or not.  Possibly PBManifest's init function should just take a session argument instead.

Reply:
ack